Update pagination style to match new designs

Blakidarkness opened the pull request:
The pagination style have been updated to match the new designs:

- `Pagination` component now works as a wrapper
- Added `totalResults` and `table` props to `Pagination`
- Added `getActualResults` function to `Pagination`
- Updated `Pagination/index`
- Updated `Pagination/styled`
- `StyledPagination` splited to `Bottom` and `Top`
- `DataTable/index` updated to new `Pagination` props

Blakidarkness committed:
Navigation Stylig

Blakidarkness committed:
Pagination top and bottom styles, datatable/index modified

cmbirk commented on src/DataTable/index.tsx:
```diff
@@ -261,6 +262,7 @@ export const DataTable = <T extends Record<string, unknown>>(
     nextPage,
     previousPage,
     setPageSize,
+    totalResults,
```
@Blakidarkness we're going to want to accept an optional incoming prop for `totalResults` that is a number. If that incoming prop is set we'll use that prop on L228 to setTotalResults instead of `data.length`. If it isn't set we can use `data.length` as the default.

The reason for this is that someone using an asynchronous pagination system would have more total results than the data currently in the table. If there are 2000 items in the API we want to set that as the total number instead of the 10 they're loading initially.

Blakidarkness committed:
added optional totalRows prop, conditional totalResults state

Blakidarkness commented:
Changes on latest commit:

- Added optional table prop `totalRows`
- Conditional state of `totalResults` takes `totalRows`  if exist, otherwise uses `data.length`
- Added `AsyncPaginationTable` story
